### PR TITLE
BIA-769 - Adding scalar functions to improve view performance

### PR DESCRIPTION
**Pull request by NSD-Ruben**
Adding scalar functions for Student History view for performance improvement

### CODE REVIEW

File: src/EdFi.AnalyticsMiddleTier.DataStandard31/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ,ssd.SchoolKey AS CurrentSchoolKey
+    ,COALESCE((CAST((DaysEnrolled - DaysAbsent) AS DECIMAL) / CAST(DaysEnrolled AS DECIMAL) * 100), 100) AS AttendanceRate
+    ,(
+        SELECT COUNT(1)
+        FROM analytics.equity_StudentDisciplineActionDim discipline
+        WHERE discipline.StudentSchoolKey = ssd.StudentSchoolKey
+        ) AS ReferralsAndSuspensions
+    ,analytics.fn_GetStudentEnrollmentHistory(ssd.StudentKey) AS EnrollmentHistory
+    ,(
+        SELECT MAX(MaxLastModifiedDate)
```
Comment (DavidJGapCR) [inline src/EdFi.AnalyticsMiddleTier.DataStandard31/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql:39]: Do we really need this select max here?

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ,ssd.SchoolKey AS CurrentSchoolKey
+    ,COALESCE((CAST((DaysEnrolled - DaysAbsent) AS DECIMAL) / CAST(DaysEnrolled AS DECIMAL) * 100), 100) AS AttendanceRate
+    ,(
+        SELECT COUNT(1)
+        FROM analytics.equity_StudentDisciplineActionDim discipline
+        WHERE discipline.StudentSchoolKey = ssd.StudentSchoolKey
+        ) AS ReferralsAndSuspensions
+    ,analytics.fn_GetStudentEnrollmentHistory(ssd.StudentKey) AS EnrollmentHistory
+    ,(
+        SELECT MAX(MaxLastModifiedDate)
```
Comment (DavidJGapCR) [inline src/EdFi.AnalyticsMiddleTier.DataStandard32/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql:39]: Do we really need this select max here?

**Comment by DavidJGapCR** [comment]
I am approving this, but if we can remove the select max statement it would be great.

**Review by jleiva-gap — APPROVED** [review]

File: src/EdFi.AnalyticsMiddleTier.DataStandard31/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql (hunk 1 above)
Comment (NSD-Ruben) [reply src/EdFi.AnalyticsMiddleTier.DataStandard31/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql]: done

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql (hunk 2 above)
Comment (NSD-Ruben) [reply src/EdFi.AnalyticsMiddleTier.DataStandard32/Equity/MSSQL/0008-View-StudentHistoryDim-Alter.sql]: done

**Pull request merged**